stream/src/main/scala/org/apache/pekko/stream/impl/fusing/Ops.scala

```diff
@@ -267,9 +266,8 @@ private[stream] object Collect {
         } catch {
           case NonFatal(ex) =>
             decider(ex) match {
-              case Supervision.Stop    => failStage(ex)
-              case Supervision.Resume  => if (!hasBeenPulled(in)) pull(in)
```

Comment:
Can we add a comment here explaining why `!hasBeenPulled(in)` is not necessary.

Reply:
Why? the Map and Filter don't have this either.

I would like to add that for in `MapConcat` operator which explain why a `!hasBeenPulled(in)` test is needed.

I think the origin reason about why this `!hasBeenPulled(in)` test is because that method is a abstract class and so that exception can happen when onPush/onPull, so it have to be guard for extending, and reduce the sub class' burden. that's not the case here.

Reply:
> Why? the Map and Filter don't have this either.

We should add the comments there as well then. Its more that when other people that are not so familiar with Akka/Pekko read the code they get a better understanding of why these exceptions are there.

Akka's problem historically is that a lot of the contributions were done within Lightbend and so there is a lot of inhouse knowledge that wasn't visible to outside contributors.

Reply:
I can't push any code because gfw blocked me, Maybe I can use a web ide?
How about we do that in a separate pr or you take care about it?

Reply:
https://pekko.apache.org/docs/pekko/current/stream/stream-customize.html
It's detailly described here.

Reply:
@mdedetrich Please unlock this, I'm not pay for everything, thanks, I just a volunteer, and this kind of information is well covered in the doc, this is code, not a book.

Reply:
@He-Pin I was away from laptop, ill just add a commit with comment onto your branch and then merge it.